### PR TITLE
Feature multiselect

### PR DESCRIPTION
**Pull request by jdmcd**
@EricKuck

### CODE REVIEW

File: Junction/CustomOption.swift [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+//
+//  Created by Jimmy McDermott on 8/10/16.
+//  Copyright © 2016 BlueLine Labs. All rights reserved.
+//
+
+import Foundation
+
+public enum CustomOption {
+    case None
+    case Custom(String)
```
Comment (EricKuck) [inline Junction/CustomOption.swift]: It's not really clear how this string will be used. Maybe do a typealias with something like TextFieldPlaceholder?

File: Junction/Junction Example/ExampleJunctionSetup.swift [hunk 2]
```diff
@@ -0,0 +1,34 @@
+//
+//  MyJunctionSetup.swift
```
Comment (EricKuck) [inline Junction/Junction Example/ExampleJunctionSetup.swift]: Header filename's out of date.

File: Junction/CustomOption.swift (hunk 1 above)
Comment (jdmcd) [inline Junction/CustomOption.swift]: Good idea

File: Junction/Junction Example/ExampleJunctionSetup.swift [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        let multipleSelectionOptions = [MultipleChoiceOption<String>(value: "google.com", isInitialValue: true), MultipleChoiceOption<String>(value: "yahoo.com", isInitialValue: false)]
+        let intMultipleSelectionOptions = [MultipleChoiceOption<Int>(value: 1, isInitialValue: true), MultipleChoiceOption<Int>(value: 2, isInitialValue: true)]
+        
+        let secondSection = StringMultipleChoiceSetting(possibleValues: multipleSelectionOptions, customOption: .Custom("Add Endpoint"), name: "Single Selection String", key: "singleSelectionString", isMultiSelect: true)
+        let thirdSection = IntMultipleChoiceSetting(possibleValues: intMultipleSelectionOptions, customOption: .None, name: "Single Selection Int", key: "singleSelectionInt", isMultiSelect: false)
+        
+        return [firstSection, secondSection, thirdSection]
+    }
+    
+    func debugMode() -> Bool {
```
Comment (EricKuck) [inline Junction/Junction Example/ExampleJunctionSetup.swift]: Maybe make this a var?

File: Junction/Junction Example/ExampleJunctionSetup.swift [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+        return true
+    }
+}
```
Comment (EricKuck) [inline Junction/Junction Example/ExampleJunctionSetup.swift]: Newline

File: Junction/Junction/IntMultipleChoiceSetting.swift [hunk 5]
```diff
@@ -14,12 +14,21 @@ public final class IntMultipleChoiceSetting: MultipleChoiceBase<Int> {
...
+        if let options = options as? [Int] {
+            if !options.contains(value) {
+                JunctionKeeper.sharedInstance.addValueToArray(key.customOption, value: value)
+                rows.append(StringSetting(placeholder: nil, defaultValue: nil, key: key, value: String(value), title: nil))
+                possibleValues.append(MultipleChoiceOption(value: value, isInitialValue: false))
+            }
+        } else {
+            JunctionKeeper.sharedInstance.addValueToArray(key.customOption, value: value)
+            rows.append(StringSetting(placeholder: nil, defaultValue: nil, key: key, value: String(value), title: nil))
+            possibleValues.append(MultipleChoiceOption(value: value, isInitialValue: false))
```
Comment (EricKuck) [inline Junction/Junction/IntMultipleChoiceSetting.swift:27]: Isn't the code here and in the if condition the same?

Comment (jdmcd) [inline Junction/Junction/IntMultipleChoiceSetting.swift:27]: Yeah it's just unwrapping it in the first part

File: Junction/Junction/Junction.swift [hunk 6]
```diff
@@ -14,12 +14,13 @@ public final class Junction {
     
     public static var settingsUpdatedBlock: SettingsCallback? = nil
     
-    public static var style: PresentationStyle!
-    public static var sections: [SectionType]!
+    public static var style: PresentationStyle! = .Shake
```
Comment (EricKuck) [inline Junction/Junction/Junction.swift]: You don't need the `!` if you're setting a default value.

Comment (jdmcd) [inline Junction/Junction/Junction.swift]: 👌

File: Junction/Junction/JunctionKeeper.swift [hunk 7]
```diff
@@ -50,26 +50,28 @@ internal final class JunctionKeeper {
...
-            return addValueForKey(key, value: [value])
-        }
-        
-        let previousValues = loadAllData()?.mutableCopy()
-        
-        previousArrayValues.addObject(value)
-        
-        let dictionary = NSDictionary(dictionary: [key: previousArrayValues])
+        let previousValues = loadAllData()?.mutableCopy()[key]
+        let previous = loadAllData()?.mutableCopy()
```
Comment (EricKuck) [inline Junction/Junction/JunctionKeeper.swift]: You probably don't need to do `loadAllData()?.mutableCopy()` twice in a row.

Comment (jdmcd) [inline Junction/Junction/JunctionKeeper.swift]: 👌

File: Junction/Junction/JunctionKeeper.swift [hunk 8]
```diff
@@ -50,26 +50,28 @@ internal final class JunctionKeeper {
...
         
-        previousValues?.addEntriesFromDictionary(dictionary as [NSObject: AnyObject])
-        
-        return previousValues!.writeToFile(path, atomically: true)
+        if let previousValues = previousValues as? NSArray {
+            let newValuesToPass = previousValues.arrayByAddingObject(value)
+            previous?.addEntriesFromDictionary([key: newValuesToPass])
+            
+            return previous!.writeToFile(path, atomically: true)
```
Comment (EricKuck) [inline Junction/Junction/JunctionKeeper.swift]: Is `previous` guaranteed to non-nil here? Why is it optional?

File: Junction/Junction/JunctionKeeper.swift [hunk 9]
```diff
@@ -87,4 +89,30 @@ internal final class JunctionKeeper {
         
         return previousValues!.writeToFile(path, atomically: true)
     }
+    
+    internal func deleteValueFromArray(key: String, valueToDelete: AnyObject) -> Bool {
+        createPlistIfNeeded()
+        
+        let filename = "JunctionData.plist"
+        let documentDirectory = NSSearchPathForDirectoriesInDomains(.DocumentDirectory, .UserDomainMask, true)[0] as String
+        let path = documentDirectory.stringByAppendingString("/\(filename)")
```
Comment (EricKuck) [inline Junction/Junction/JunctionKeeper.swift]: You should probably have all the code that gets the path to the plist in one place. Guessing these same 3 lines are in a few places throughout the code base.

File: Junction/StringExtensions.swift [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+//  StringExtensions.swift
+//  Junction
+//
+//  Created by Jimmy McDermott on 8/10/16.
+//  Copyright © 2016 BlueLine Labs. All rights reserved.
+//
+
+import Foundation
+
+extension String {
```
Comment (EricKuck) [inline Junction/StringExtensions.swift]: This should be internal.

File: Junction/StringExtensions.swift [hunk 11]
```diff
@@ -0,0 +1,23 @@
...
+    var customOption: String {
```
Comment (EricKuck) [inline Junction/StringExtensions.swift]: Kind of confusing naming. Just looking at this, I have no idea what it does.

**Pull request merged**